tests/test_tasks.py

```diff
@@ -2365,6 +2432,31 @@ def initial_task():
         assert task_with_options.retries == 0
         assert task_with_options.retry_delay_seconds == 0
 
+    def test_with_options_refresh_cache(self):
+        @task(cache_key_fn=lambda *_: "cache hit")
+        def foo(x):
+            return x
+
+        @flow
+        def bar():
+            return (
+                foo._run(1),
+                foo._run(2),
+                foo.with_options(refresh_cache=True)._run(3),
+                foo._run(4),
```

Comment:
Note that `_run` is a holdover from before and we encourage calls with the `return_state` option instead for new tests so `foo(1, return_state=True)`.